[strings] Remove duplicated "avoid" strings

dvdmrtnz opened the pull request:
As mentioned by @RicoElectrico in https://github.com/organicmaps/organicmaps/pull/6251#issuecomment-1751695032

biodranik commented:
Where they were used/visible?

dvdmrtnz commented:
> Where they were used/visible?

When there is a toll road in the way, and you get this red bubble. If you click on it:

<img src="https://github.com/organicmaps/organicmaps/assets/47610359/79ecea51-cfba-4f5e-8a6a-396317bec313" width="350"/>

pastk commented on data/strings/strings.txt:
```diff
@@ -22429,135 +22429,6 @@
     zh-Hans = 轮渡
     zh-Hant = 渡輪
 
-  [avoid_toll_roads_placepage]
```
I think it makes sense to use these translations in the new string IDs like `[avoid_tolls]`. Because these translations had been in the repo for a long time and were proof-read by translators for sure, unlike most of the new automatic translations.

biodranik replied:
Aren't they different?

pastk replied:
True, those strings are "Avoid toll roads", but the new ones are "Avoid tolls".
But "Avoid unpaved roads" and "Avoid ferry crossings" are the same.

pastk commented:
Ouch! there seems to be another set of very similar strings, e.g.

```
  [avoid_tolls_carplay_1]
    tags = ios
    en = No toll road

  [avoid_tolls_carplay_2]
    tags = ios
    en = Avoid toll road

  [avoid_unpaved_carplay_1]
    tags = ios
    en = No dirt road

  [avoid_unpaved_carplay_2]
    tags = ios
    en = Avoid unpaved

  [avoid_ferry_carplay_1]
    tags = ios
    en = No ferries

  [avoid_ferry_carplay_2]
    tags = ios
    en = Avoid ferries
```

dvdmrtnz commented:
* I have removed the duplicated CarPlay strings too
* I have replaced the automated translations in `avoid_unpaved` and `avoid_ferry` with the existing strings from `avoid_unpaved_roads_placepage` and `avoid_ferry_crossing_placepage` except for FR, DE, ES, PT, PT-BR, RU and UK languages which were reviewed in the previous PR and should be okay

biodranik commented:
How new strings look on CarPlay? Are they cut as in AA?

dvdmrtnz commented:
> How new strings look on CarPlay? Are they cut as in AA?

Some longer strings get cut

![Simulator Screenshot - iPhone 13 mini - 2023-10-08 at 22 05 39](https://github.com/organicmaps/organicmaps/assets/47610359/cc07c9a2-4e31-455a-bbfd-8429bc9cc0e4)
![Simulator Screenshot - iPhone 13 mini - 2023-10-08 at 22 12 25](https://github.com/organicmaps/organicmaps/assets/47610359/d268ce45-374b-44cf-b99b-94aba6aa0f75)

dvdmrtnz commented:
I think a lot of strings could be shortened because in some languages they seem to be using longer forms instead of shorter ones that are used in some other languages:

- Avoid toll roads -> Avoid tolls
- Avoid ferry crossings -> Avoid ferries
- Avoid unpaved roads -> Avoid unpaved

biodranik commented:
Why can't we use shorter forms? Were previously shorter forms used in other languages?

Looks like there is a limit of approx 25 chars in CarPlay now, it may make sense to mention it in the comment, and also mention max chars for AA.

biodranik commented:
And thanks for the screenshots. We need to create an UX issue for CarPlay, it's not clear when avoiding is enabled or disabled :)

dvdmrtnz commented:
I have tried to take the available existing short strings from CarPlay. Some languages still need help. If some of our translators can help to make them shorter it would be great.

Bulgarian:
![](https://github.com/organicmaps/organicmaps/assets/47610359/c0616380-ca6f-4dd2-b660-9b6a06096338)

Czech:
![](https://github.com/organicmaps/organicmaps/assets/47610359/bf996bbe-bf5e-4815-b208-5962f7e13564)

Basque:
![](https://github.com/organicmaps/organicmaps/assets/47610359/f4b39518-517c-4649-9fb6-784ff64d2890)

Italian:
![](https://github.com/organicmaps/organicmaps/assets/47610359/7d3fb553-03dc-4a6e-91a5-ecc47a15b335)

Russian:
![](https://github.com/organicmaps/organicmaps/assets/47610359/3eea2724-2a23-474d-9671-b87b2192878d)

Slovak:
![](https://github.com/organicmaps/organicmaps/assets/47610359/22a01903-daad-43d2-b4a9-bc8e00dfbcee)

Ukranian:
![](https://github.com/organicmaps/organicmaps/assets/47610359/2d37831a-ae48-4871-83a3-73725a4395ed)

matheusgomesms commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
+    nb = Unngå uasfalterte veier
+    nl = Aardewegen vermijden
+    pl = Unikaj dróg gruntowych
+    pt = Evitar vias não pavimentadas
+    pt-BR = Evitar vias não pavimentadas
```
```suggestion
    pt = Evitar não-pavimentadas
    pt-BR = Evitar não-pavimentadas
```

matheusgomesms commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
-    nl = Vermijd overtochten met de veerboot
+    nl = Ferries vermijden
     pl = Unikaj przepraw promowych
     pt = Evitar balsas
```
```suggestion
    pt = Evitar ferribotes
```

dvdmrtnz replied:
Thanks!

matheusgomesms reviewed (changes requested):
Shortened some strings in PT and PT-BR

jimcarst commented on data/strings/strings.txt:
```diff
@@ -21967,16 +21967,17 @@
+    el = Αποφυγή τα διόδια
     es = Evitar peajes
     et = Vältige tasulisi teid
     eu = Saihestu ordainpeko errepideak
```
```suggestion
    eu = Saihestu ordainpeko errepideak
    nl = Vermijd tolwegen
```

jimcarst commented on data/strings/strings.txt:
```diff
@@ -22096,7 +22102,7 @@
+    ca = Evita autopistes
     cs = Vyhněte se dálnici
     da = Undgå motorvejen
     de = Autobahnen vermeiden
```
```suggestion
    de = Autobahnen vermeiden
    nl = Vermijd snelwegen
```

dvdmrtnz commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
+    lt = Vengti neasfaltuotų kelių
+    mr = कच्चे मार्ग टाळा
+    nb = Unngå uasfalterte veier
+    nl = Aardewegen vermijden
```
Thanks!

biodranik commented on data/strings/strings.txt:
```diff
@@ -21967,16 +21967,17 @@
     zh-Hant = 繞行設定
 
   [avoid_tolls]
+    comment = Should be maximum 25 characters
```
comment = Recommended length for CarPlay and Android Auto is around 25-27 characters.

@AndrewShkrob can you please confirm the AA limit?

AndrewShkrob replied:
Approximately 26 characters, but it depends on word length and line breaks.

biodranik commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
-    be = Пазбягаць грунтавыя дарогі
+    af = ermy grondpaaie
+    ar = تجنب الطرق الغير معبدة
+    be = Пазбягаць грунтовых дарог
```
be = Пазбягаць грунтавыя дарогі

biodranik commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
+    en = Avoid ferries
+    af = Vermy veerbootkruisings
+    ar = تجنب استخدام العبّارات
+    be = Пазбягаць паромаў
```
be = Пазбягаць паромы

biodranik commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
     pt-BR = Evitar balsas
-    ro = Evitați traversările cu feribotul
+    ro = Evită trecerile cu bac
     ru = Избегать паромных переправ
```
ru = Избегать паромы

dvdmrtnz replied:
Thanks!

![](https://github.com/organicmaps/organicmaps/assets/47610359/526b791a-97ce-467c-8c9d-69539fd93c2f)

biodranik commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
+    pt = Evitar não-pavimentadas
+    pt-BR = Evitar não-pavimentadas
+    ro = Evită drumurile neasfaltate
     ru = Избегать грунтовых дорог
```
ru = Избегать грунтовые дороги

biodranik commented on data/strings/strings.txt:
```diff
@@ -22008,86 +22009,91 @@
+    sw = Epuka barabara za vumbi
+    th = เลี่ยงถนนดินทั้งหมด
+    tr = Asfaltsız yollardan kaçın
     uk = Уникати ґрунтових доріг
```
@pastk

pastk replied:
??

biodranik commented on data/strings/strings.txt:
```diff
@@ -22115,7 +22121,7 @@
+    nl = Vermijd snelwegen
     pl = Unikaj autostrad
     pt = Evitar autoestradas
     pt-BR = Evitar rodovias
```
ru = Избегать автомагистрали

dvdmrtnz commented:
@pastk Can you suggest a string shorter than "Уникати ґрунтових доріг" to say "Avoid ferries" in Ukranian? So it doesn't get cut off in CarPlay:

![](https://github.com/organicmaps/organicmaps/assets/47610359/d5858ede-9e6d-496a-bd10-87cf21110faf)

pastk commented:
Sorry I'm not an Ukrainian speaker actually (sometimes I can fix very obvious things).
@strump could you help please?

dvdmrtnz commented:
@strump Would "Уникати поромів" be okay for "Avoid ferries"?

rtsisyk commented:
It would be better to use positive booleans, i.e. "Use ferries" and "Use paved roads" instead of "Avoid ferries" and "Avoid paved roads".

Here is a relevant article from the Google's restroom:

![image](https://github.com/organicmaps/organicmaps/assets/1799054/ae1c4a65-de13-4320-97c6-889478940174)

dvdmrtnz commented:
> It would be better to use positive booleans, i.e. "Use ferries" and "Use paved roads" instead of "Avoid ferries" and "Avoid paved roads".
> 
> Here is a relevant article from the Google's restroom:
> 
> ![image](https://user-images.githubusercontent.com/1799054/280937344-ae1c4a65-de13-4320-97c6-889478940174.png)

Use ferries is not the same as not avoid ferries. See https://github.com/organicmaps/organicmaps/issues/5779

biodranik commented on data/strings/strings.txt:
```diff
@@ -21965,86 +21966,91 @@
     tags = android,ios
     en = Avoid unpaved roads
-    ar = تجنب الطرق الترابية
+    af = ermy grondpaaie
```
Ermy?

dvdmrtnz replied:
I think copypasting somewhere the "V" was lost. Correct translation seems to be "Vermy grondpaaie". Fixed

biodranik commented on iphone/Maps/Classes/CarPlay/Template Builders/SettingsTemplateBuilder.swift:
```diff
@@ -37,8 +36,7 @@ final class SettingsTemplateBuilder {
     if options.avoidDirty { unpavedIconName += "_active" }
-    let unpavedButton = CPGridButton(titleVariants: [L("avoid_unpaved_carplay_1"),
-                                                     L("avoid_unpaved_carplay_2")],
+    let unpavedButton = CPGridButton(titleVariants: [L("avoid_unpaved")],
```
@AndrewShkrob CarPlay supports several titles for different sizes/lengths. Is it also supported in AA?

biodranik reviewed (commented):
Please remove the regenerated commit, rebase on the latest master, and then regenerate strings again.

dvdmrtnz commented:
Rebased and regenerated

biodranik reviewed (approved):
Thanks!
